idx/cassandra/cassandra.go

```diff
@@ -266,8 +266,14 @@ func (c *CasIdx) AddOrUpdate(data *schema.MetricData, partition int32) {
 	}
 
 	if updateIdx {
+		// set the new metricdef based on the data, however if this data will go back in time
+		// it'll be rejected data, and we should stick to the higher LastUpdate we already had.
 		def := schema.MetricDefinitionFromMetricData(data)
+		if inMemory && existing.LastUpdate > def.LastUpdate {
```

Comment:
If we already know at this point that the metric will be rejected once we try adding it, shouldn't we just return an error here and then make the `input.DefaultHandler.Process()` abort instead of letting it continue?